gemrb/core/Map.cpp

```diff
@@ -3057,56 +3017,84 @@ bool Map::IsVisible(const Point &pos, int explored)
 	return (VisibleBitmap[by] & bi)!=0;
 }
 
+
 //point a is visible from point b (searchmap)
-bool Map::IsVisibleLOS(const Point &s, const Point &d) const
-{
-	int sX=s.x/16;
-	int sY=s.y/12;
-	int dX=d.x/16;
-	int dY=d.y/12;
+bool Map::IsVisibleLOS(const Point &s, const Point &d, bool checkWalkability) const {
+	int sX = s.x / 16;
+	int sY = s.y / 12;
+	int dX = d.x / 16;
+	int dY = d.y / 12;
 	int diffx = sX - dX;
 	int diffy = sY - dY;
 
 	// we basically draw a 'line' from (sX, sY) to (dX, dY)
 	// we want to move along the larger axis, to make sure we don't miss anything
-	if (abs( diffx ) >= abs( diffy )) {
+	if (abs(diffx) >= abs(diffy)) {
 		// (sX - startX)/elevationy = (sX - startX)/fabs(diffx) * diffy
-		double elevationy = std::fabs((double)diffx ) / diffy;
+		double elevationy = std::fabs((double) diffx) / diffy;
 		if (sX > dX) {
 			// right to left
 			for (int startx = sX; startx >= dX; startx--) {
 				// sX - startx >= 0, so subtract (due to sign of diffy)
 				//if (GetBlocked( startx, sY - ( int ) ( ( sX - startx ) / elevationy ) ) & PATH_MAP_NO_SEE)
-				if (GetBlocked( startx, sY - ( int ) ( ( sX - startx ) / elevationy ) ) & PATH_MAP_SIDEWALL)
+				unsigned int blockStatus = GetBlocked(startx,
+											  sY - (int) ((sX - startx) / elevationy));
+				if (blockStatus & PATH_MAP_SIDEWALL)
 					return false;
+				if (checkWalkability) {
+					if (blockStatus != PATH_MAP_PASSABLE &&
+						blockStatus != PATH_MAP_TRAVEL)
```

Comment:
Another case where they'll nicely align only at 4 spaces per tab. Just leave it in one line.